modules/etcd-creating-automated-backups.adoc

```diff
@@ -0,0 +1,320 @@
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - <example-master-node> <.>
+----
+<.> The amount of storage available to the PV. Adjust this value for your requirements.
+<.> Replace this value with the node to attach this PV to.
++
+[TIP]
+====
+Run the following command to list the available nodes:
+
+[source,terminal]
+----
+$ oc get nodes
+----
+====
+
+.. Verify the creation of the PV by running the following command:
++
+[source,terminal]
+----
+$ oc get pv
+----
++
+.Example output
+[source,terminal]
+----
+NAME                    CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS    REASON   AGE
+etcd-backup-pv-fs       100Gi      RWX            Delete           Available           local-storage            10s
+----
+
+.. Create a PVC named `etcd-backup-pvc.yaml` with contents such as the following:
++
+[source,yaml]
+----
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: etcd-backup-pvc
+spec:
+  accessModes:
+  - ReadWriteMany
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 10Gi <.>
+  storageClassName: local-storage
+----
+<.> The amount of storage available to the PVC. Adjust this value for your requirements.
+
+.. Apply the PVC by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f etcd-backup-pvc.yaml
+----
+
+. Create a custom resource definition (CRD) file named `etcd-recurring-backups.yaml`. The contents of the created CRD define the schedule and retention type of automated backups.
++
+For the default retention type, use contents such as the following:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1alpha1
+kind: Backup
+metadata:
+  name: etcd-recurring-backup
+spec:
+  etcd:
+    schedule: "20 4 * * *" <.>
+    timeZone: "UTC"
+    retentionPolicy:
+      retentionType: "" <.>
+    pvcName: etcd-backup-pvc
+----
+<.> The `CronTab` schedule for recurring backups. Adjust this value for your needs.
+<.> The `""` retention type specifies the default type and size.
++
+To use retention based on the maximum number of backups, add the following key-value pairs to the `retentionPolicy`:
++
+[source,yaml]
+----
+retentionPolicy:
+  retentionType: RetentionNumber
+  retentionNumber:
+    maxNumberOfBackups: 5 <.>
+----
+<.> The maximum number of backups to retain. Adjust this value for your needs.
++
+[WARNING]
+====
+A known issue causes the number of retained backups to be one greater than the configured value.
+====
++
+For retention based on the file size of backups, use the following:
++
+[source,yaml]
+----
+retentionPolicy:
+  retentionType: RetentionSize
+  retentionSize:
+    maxSizeOfBackupsGb: 10 <.>
+----
+<.> The maximum file size of the retained backups in gigabytes. Adjust this value for your needs.
++
+[WARNING]
+====
+A known issue causes the maximum size of retained backups to be up to 10 gigabytes greater than the configured value.
```

Comment:
It's actually by design - also because we don't know the size of the backups up front. But we can leave this "known issue" here for the time being.

Reply:
You referred to this earlier as a "known bug," so I'm honestly confused here. Is it a bug or not? If it's not a bug, then I feel like the name `maxSizeOfBackups` is misleading, as it's expected that the actual maximum size will be larger than the maximum size that the user configures.

This seems very strange to me.

Reply:
> maxSizeOfBackups is misleading, as it's expected that the actual maximum size will be larger than the maximum size that the user configures.

it is to some degree, that's why I wanted you to document it before customers open dozens of bug tickets around this.

Reply:
Okay, that makes sense. Thanks for explaining!